woocommerce/payment-gateway/class-sv-wc-payment-gateway-direct.php

```diff
@@ -746,21 +747,21 @@ protected function do_credit_card_transaction( $order, $response = null ) {
 			// credit card order note
 			if ( $account_number ) {
 				$message = sprintf(
-					/* translators: Placeholders: %1$s - payment method title, %2$s - environment ("Test"), %3$s - transaction type (authorization/charge), %4$s - card type (mastercard, visa, ...), %5$s - last four digits of the card */
+					/* translators: Example: "Braintree Test Authorization Approved: Mastercard ending in 1234". Placeholders: %1$s - payment method title, %2$s - environment ("Test"), %3$s - transaction type (authorization/charge), %4$s - card type (mastercard, visa, ...), %5$s - last four digits of the card */
 					esc_html__( '%1$s %2$s %3$s Approved: %4$s ending in %5$s', 'woocommerce-plugin-framework' ),
 					$this->get_method_title(),
-					$this->is_test_environment() ? esc_html_x( 'Test', 'noun, software environment', 'woocommerce-plugin-framework' ) : '',
-					$this->perform_credit_card_authorization( $order ) ? esc_html_x( 'Authorization', 'credit card transaction type', 'woocommerce-plugin-framework' ) : esc_html_x( 'Charge', 'noun, credit card transaction type', 'woocommerce-plugin-framework' ),
+					$this->is_test_environment() ? esc_html_x( 'Test', 'Noun, software environment', 'woocommerce-plugin-framework' ) : '',
+					$this->perform_credit_card_authorization( $order ) ? esc_html_x( 'Authorization', 'Credit card transaction type', 'woocommerce-plugin-framework' ) : esc_html_x( 'Charge', 'noun, credit card transaction type', 'woocommerce-plugin-framework' ),
 					SV_WC_Payment_Gateway_Helper::payment_type_to_name( $card_type ),
 					$last_four
 				);
 			} else {
 				$message = sprintf(
-					/* translators: Placeholders: %1$s - payment method title, %2$s - environment ("Test"), %3$s - transaction type (authorization/charge), %4$s - card type (mastercard, visa, ...) */
+					/* translators: Example: "Authorize.Net Test Charge Approved: Mastercard" - Placeholders: %1$s - payment method title, %2$s - environment ("Test"), %3$s - transaction type (authorization/charge), %4$s - card type (mastercard, visa, ...) */
 					esc_html__( '%1$s %2$s %3$s Approved: %4$s', 'woocommerce-plugin-framework' ),
```

Comment:
Not a comment on the current changes, but something like this is extremely difficult to translate, because the placeholders are also translatable and may ned to use different inflections based on the context or the grammatical case used...

For example, in Estonian this sentece will become something like:
```
Authorize.net Testkeskkond Kaardimakse vastu võetud: Mastercard
```
But it's wrong, it should be (no capitalization, different suffix/inflection for "testkeskkond") - same [as here](https://x.com/paragulka/status/1702301115163807980?s=20).
```
Authorize.net testkeskkonna kaardimakse vastu võetud: Mastercard
```

It's a difficult problem to solve with gettext, though - if not impossible, because gettext does not support much beyond pluralization rules. The most robust solution would be to avoid using placeholders, and always write out full sentences (not something I'm actually recommending to do at the moment though).

Another option is to use something like Fluent or MessageFormat, which have much better support for inflections, cases, etc - but they are foreign to the WP ecosystem, unfortunately.

Reply:
in the end this is a quasi-machine message, more of a notification for the merchant running a test so I think we can accept some degree of linguistic inaccuracy